### PR TITLE
Prepare 6.0.4 release

### LINKED ISSUE #431: [Feature Request] Add copy session url to clipboard functionality
I think it would be awesome to have an option to copy the currently watched session URL to clipboard, for easy sharing with colleagues. 🙃

There's an action sheet, but some services don't support it. For example, Slack.

Thanks! ✌️

### LINKED ISSUE #464: App hangs for several seconds at launch
We have been getting reports of this issue since last year and @allenhumphreys tracked it down to an issue with Realm's linking objects, which was reported to Realm [here](https://github.com/realm/realm-cocoa/issues/5816). We are currently waiting for a response from Realm to determine our course of action.

### LINKED ISSUE #468: Featured tab related crash when playing a video
```
Crashing on exception: Cannot register notification blocks from within write transactions.
```

[AnnotatedStackTrace.txt](https://github.com/insidegui/WWDC/files/2109519/AnnotatedStackTrace.txt)

Not sure what the problem is exactly but it definitely exists and should block releasing the featured tab.

### LINKED ISSUE #475: Transcript tab is empty for a specific session
After integrating 2017/2018 transcripts from ASCIIWWDC, I noticed that the session "Vision with Core ML" from 2018 shows the "Transcript" tab, but it's empty. I haven't looked thoroughly enough but this could be happening to other sessions as well. We need to determine if this is an issue with the logic used to determine whether to display the transcript tab or an issue with the transcript itself.

screen shot 2018-06-19 at 12 15 40

### LINKED ISSUE #479: Player fails to hide, obscuring the image/play button of selected sessions
Introduced by #428 , we're trying to achieve the correct balance of player hiding.

This bug results in the player being visible on sessions other than the currently playing session.

### PR DESCRIPTION
**Pull request by allenhumphreys**
#### Fixes
Switch player timestamps to monospace system font
Fixes related sessions navigation so that it can expose the destination session
Fixes application hanging by updating Realm
Fixes fullscreen player hiding incorrectly when browsing sessions
Optimizes drawing of WWDCTableRowView
Fixes application hangs by asynchronously accessing video information
Fixes non-editable text swallowing key events
Fixes cursor hiding when it shouldn't involving fullscreen
Switch to https for transcripts
Fixes empty transcripts being treated as real transcripts
Enabled subtitles for 2017 and 2018

#### Small Enhancements
Player buttons now accept first mouse, allowing play/pause while not frontmost
Add support for seeking videos using keyboard arrows
Defines option-click behaviour when toggling speed
Share via copy URL to clipboard

<details><summary>All Commits</summary>
<p>

Ensure player (un)hides correctly on selection changes, fixes #479 (#480)  
Update comment to explain why the function is empty  
Optimizes drawing of WWDCTableRowView  
Read the docs, fixed subtitle options loading  
Cleanup replaced code, show loading spinner after delay  
More robust spinner showing, error handling, some polish  
Asynchronously access video information to avoid hangs, preliminary spinner  
Further contributing updates  
Bug fix (#463): When tapping on non-editable text, player shortcut events get swallowed  
Addresses player controls hiding while still preventing disappearing cursor  
Use https for asciiwwdc  
Treat empty transcript as nothing, fixes #475  
Added section to readme pointing to contribution guidelines  
Updated contribution guidelines  
Fixes #464 by updating Realm to 3.7.3  
Player buttons now accept first mouse, allowing play/pause while not frontmost  
Fixes #468 crash and session display issue  
Updated comment  
Enabled indexing of transcripts for 2017 and 2018  
Switch player timestamps to monospace system font  
Fixes small retain cycle  
Add RelatedResourceType to account for "download" resources  
Remove configuration flag since it was fully implemented  
Respect user preference when seeking using keyboard arrows  
Add support for seeking videos using keyboard arrows  
Rename identifiers  
Defines option-click behaviour when toggling speed  
Consolidate frequently replicated logic, clear results on intial row display  
Fixes filter clearing behaviour during selection while navigating  
Fixes session lists to allow arbitrary navigation  
Fixes reported layout ambiguity  
Fixes mouse hiding when it shouldn't  
Correctly put the string on the paste board  
Copy as a string to allow for more flexible use  
Add asset for copy share service  
Closes #431, copy URL to clipboard  
Remove duplicated tooltips  
Fixes 418, player is hidden at an inopportune time  
Typo  
</p>
</details>

### CODE REVIEW

**Comment by insidegui** [comment]
Release notes:

<h3>Our "iOS 12" release</h3>
<p>WWDC came and went and a huge topic, especially for iOS 12, was performance and stability. Inspired by that, we're releasing WWDC for macOS 6.0.4, with the following fixes:</p>
<ul>
<li>Player timestamps now use the correct monospace system font</li>
<li>Optimized the UI for faster iteractions</li>
<li>Fixed related sessions navigation so that it can expose the destination session</li>
<li>Fixed a hang that would occur sometimes when the app was loading its data</li>
<li>Fixed fullscreen player hiding incorrectly when browsing sessions</li>
<li>Fixed a hang that could occur when starting playback of a streaming video</li>
<li>Fixed an issue that would prevent the space bar from playing/pausing the video</li>
<li>Fixed an issue where the cursor would be hidden when switching spaces while playing in fullscreen</li>
<li>Transcripts from ASCIIWWDC are now downloaded over HTTPS</li>
</ul>
<h4>We also made some enhancements:</h4>
<ul>
<li>Enabled subtitles for 2017 and 2018 (thanks Mattt)</li>
<li>The player now allows play/pause while the app is not the frontmost app</li>
<li>It's now possible to seek using the keyboard arrow keys</li>
<li>Option-clicking the speed button on the player will go to the previous playback speed</li>
<li>In the share menu, there's now an option to copy the link for a session to the clipboard</li>
</ul>

**Comment by insidegui** [comment]
@allenhumphreys Please do not merge this until we're sure that issue we talked about on Slack was just a one time thing and I have added the actual release zip to the PR 😅

File: Releases/appcast_v5.xml [hunk 1]
```diff
@@ -6,6 +6,46 @@
...
+					<li>In the share menu, there's now an option to copy the link for a session to the clipboard</li>
+				</ul>
+			]]></description>
+            <pubDate>Sat, 30 Jun 2018 15:00:00 -0300</pubDate>
+            <enclosure url="https://github.com/insidegui/WWDC/releases/download/6.0.4/WWDC_v6.0.4.zip" sparkle:version="821" sparkle:shortVersionString="6.0.4" length="16090272" type="application/octet-stream" />
+            <sparkle:minimumSystemVersion>10.12.2</sparkle:minimumSystemVersion>
+        </item>		
+		
+		<item>
+            <title>Version 6.0.2</title>
```
Comment (allenhumphreys) [inline Releases/appcast_v5.xml]: @insidegui copy pasta?

Comment (insidegui) [reply Releases/appcast_v5.xml]: Hahahahaha yeap

Comment (allenhumphreys) [reply Releases/appcast_v5.xml]: Haha, no @insidegui the number wasn’t wrong, that whole item entry was duplicated!

**Pull request merged**